### ISSUE TITLE
Submission: refnet

### ISSUE THREAD

**Issue by aurielfournier** [ropensci/software-review#256]
### Summary

-   What does this package do? (explain in 50 words or less):
`refnet` is a package to read, organize, geocode, analyze, and visualize Clarivate Web of Knowledge/Web of Science, format reference data files for scientometric, social network, and Science of Science analyses.

-   Paste the full DESCRIPTION file inside a code block below:

```
Package: refnet
Type: Package
Title: Thomson Reuters Web of Knowledge/Science and ISI Reference Data Tools
Version: 0.6
Date: 2018-08-26
Authors@R: c(person("Auriel M.V. Fournier", "Developer", role = c("aut"),
                     email = "aurielfournier@gmail.com"),
              person("Forrest R. Stevens", "Developer", role = "aut"),
              person("Matthew E. Boone", "Developer", role = "aut"),
              person("Emilio Bruna", "Developer", role=c("aut","cre"), 
              email="embruna@ufl.edu"))
Maintainer: Emilio Bruna <embruna@ufl.edu>
Description: This function reads Thomson Reuters Web of Knowledge/Science and ISI format reference data files into an R friendly data format and can optionally write the converted data to a friendly CSV format.
License: GPL-3
Imports: maptools, maps, rworldmap, RecordLinkage, Matrix, igraph, network, sna, Hmisc, ggplot2, stringi, stringr, ggmap, Rdpack, tidyr, dplyr, tibble
RoxygenNote: 6.1.0
RdMacros: Rdpack
Suggests: testthat, utils
VignetteBuilder: utils
Encoding: UTF-8
```

-   URL for the package (the development repository, not a stylized html page):
https://github.com/embruna/refnet

- Please indicate which category or categories from our [package fit policies](https://github.com/ropensci/onboarding/blob/master/policies.md#package-fit) this package falls under ***and why**(? (e.g., data retrieval, reproducibility. If you are unsure, we suggest you make a pre-submission inquiry.):

Data extraction and munging, since it takes data from one format, and transforms it into something that is useful, and also matches up records among authors. 

[Note, the link for the package fit, does not lead to that page anymore, and I couldn't find anything about package fit in the linked policies]

-   Who is the target audience and what are scientific applications of this package?  

Scientists interested in studying the networks of a particular author, subject area or journal. 

-   Are there other R packages that accomplish the same thing? If so, how does
yours differ or meet [our criteria for best-in-category](https://github.com/ropensci/onboarding/blob/master/policies.md#overlap)?

-   If you made a pre-submission enquiry, please paste the link to the corresponding issue, forum post, or other discussion, or @tag the editor you contacted.

https://github.com/ropensci/onboarding/issues/247

### Requirements

Confirm each of the following by checking the box.  This package:

- [X] does not violate the Terms of Service of any service it interacts with. 
- [X] has a CRAN and OSI accepted license.
- [X] contains a README with instructions for installing the development version. 
- [X] includes documentation with examples for all functions.
- [X] contains a vignette with examples of its essential functions and uses.
- [X] has a test suite.
- [X] has continuous integration, including reporting of test coverage, using services such as Travis CI, Coveralls and/or CodeCov.
- [X] I agree to abide by [ROpenSci's Code of Conduct](https://github.com/ropensci/onboarding/blob/master/policies.md#code-of-conduct) during the review process and in maintaining my package should it be accepted.

#### Publication options

- [yes] Do you intend for this package to go on CRAN?  
- [yes] Do you wish to automatically submit to the [Journal of Open Source Software](http://joss.theoj.org/)? If so:
    - [X] The package has an **obvious research application** according to [JOSS's definition](http://joss.theoj.org/about#submission_requirements).
    - [X] The package contains a `paper.md` matching [JOSS's requirements](http://joss.theoj.org/about#paper_structure) with a high-level description in the package root or in `inst/`.
    - [not yet] The package is deposited in a long-term repository with the DOI: 
    - (*Do not submit your package separately to JOSS*)
- [no] Do you wish to submit an Applications Article about your package to [Methods in Ecology and Evolution](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/)? If so:
    - [ ] The package is novel and will be of interest to the broad readership of the journal. 
    - [ ] The manuscript describing the package is no longer than 3000 words.
    - [ ] You intend to archive the code for the package in a long-term repository which meets the requirements of the journal (see [MEE's Policy on Publishing Code](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/journal-resources/policy-on-publishing-code.html))
    - (*Scope: Do consider MEE's [Aims and Scope](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/aims-and-scope/read-full-aims-and-scope.html) for your manuscript. We make no guarantee that your manuscript will be within MEE scope.*)
    - (*Although not required, we strongly recommend having a full manuscript prepared when you submit here.*)
    - (*Please do not submit your package separately to Methods in Ecology and Evolution*)

### Detail

- [yes] Does `R CMD check` (or `devtools::check()`) succeed?  Paste and describe any errors or warnings:

- [yes] Does the package conform to [rOpenSci packaging guidelines](https://github.com/ropensci/onboarding/blob/master/packaging_guide.md)? Please describe any exceptions:

- If this is a resubmission following rejection, please explain the change in circumstances:

- If possible, please provide recommendations of reviewers - those with experience with similar packages and/or likely users of your package - and their GitHub user names:

Heather Piwowar @hpiwowar

**Comment by sckott** [comment]
Thanks very much for your submission @aurielfournier - we're discussing now and will get back to you soon

**Comment by maelle** [comment]
Thanks for your submission @aurielfournier! I see the package doesn't have any test and doesn't have continuous integration yet. I suggest we put the submission on hold while you sort that, unless you can and want to add this within a week or so? There is some guidance in [this guide](https://ropensci.github.io/dev_guide/), and I am happy to answer any question here or via Slack!

I was also looking at the dependencies, there are many of them in DESCRIPTION and
* at least one doesn't seem used at all in the package (`stringi`)
* it seems you have written the NAMESPACE by hand? We'd prefer you to use `@importFrom` (or even just `pkg::fun` when calling the function) and `@export` tags in R files. I wouldn't recommend importing whole packages unless needed. See http://r-pkgs.had.co.nz/namespace.html#imports

**Comment by aurielfournier** [comment]
Thanks @maelle I and my co-authors are working on it, but its taken a bit longer then we expected. Appreciate your patience!

**Comment by maelle** [comment]
No problem, and happy to help if I/we can!

**Comment by aurielfournier** [comment]
@maelle package now has tests and continuous integration.

I removed stringi from the DESCRIPTION file. 

I have fixed the issues from the NAMESPACE file.

Huge thanks to my coauthor @birderboone for doing the heavy lifting to get this over the finish line!

I think we are ready for review now. If you have any other things that need to be addressed let me know.

Thanks!

**Comment by maelle** [comment]
:wave: @aurielfournier @biderboone! Awesome, thanks to both of you! A few comments before I do the last editor checks:
* Could you please add the Travis badge at the top of the README?
* The Travis build is failing. To fix this (both WARNING and NOTEs)   
    * "Undocumented arguments in documentation object 'references_read'
  ‘include_all’" Update the docs
    * prefix `utils` functions e.g. `utils::flush.console()`
    * For some _variables_ you might need to create such a file: https://github.com/ropensci/opencage/blob/master/R/globalVariables.R to define them as global variables (the ones that appear in NOTEs). Unless they come up in `dplyr` function in which case you can write e.g. `mutate(df, y = .data$a + .data$x)` to make the NOTE disappear see [this vignette](https://cran.r-project.org/web/packages/dplyr/vignettes/programming.html)
    * Reg JOSS in `rtimicropem` I put the paper in a paper/ folder https://github.com/ropensci/rtimicropem/tree/master/paper and buildignored it `usethis::use_build_ignore("paper/")` or something like that.
* Once the Travis build passes please add a code coverage report. Run `usethis::use_coverage("codecov")` which will give you stuff to add to the Travis config file, browse codecov maybe (I can't remember) and give you the code to paste in the README to get a badge.
* I see two issues about postal codes (leading zeros, Brazil code), do they need to be fixed before review?
* Regarding the other issues I admire the naming scheme. Just in case you ignored that, GitHub also has a milestone feature: https://help.github.com/articles/about-milestones/
* Please add a code of conduct and contributing guide (this can be extremely short) https://ropensci.github.io/dev_guide/collaboration.html#friendlyfiles No need to add the PR and issue templates yet.

**Comment by maelle** [comment]
Yay, green badge! Can you also add a coverage badge? Run `usethis::use_coverage("codecov")` which will give you stuff to add to the Travis config file, browse codecov maybe (I can't remember) and give you the code to paste in the README to get a badge.

**Comment by aurielfournier** [comment]
Done! Sorry I missed that.

**Comment by maelle** [comment]
Thank you! A few more things before I search for reviewers (then they and you have less work :wink:).

- [x] **Fit**: The package meets criteria for [fit](policies.md#fit) and [overlap](policies.md#fit)
- [x] **Automated tests:** Package has a testing suite and is tested via Travis-CI or another CI service.
- [x] **License:** The package has a CRAN or OSI accepted license
- [x] **Repository:** The repository link resolves correctly
- [ ] **Archive** (JOSS only, may be post-review): The repository DOI resolves correctly
- [ ] **Version** (JOSS only, may be post-review): Does the release version given match the GitHub release (v1.0.0)?

---

#### Editor comments

* `goodpractice` output

```r
✖ write short and simple functions. These
    functions have high cyclomatic complexity:authors_clean
    (68).
```

Maybe you can split it in several helper functions?

```r
  ✖ omit "Date" in DESCRIPTION. It is not required
    and it gets invalid quite often. A build date will be
    added to the package when you perform `R CMD build` on it.
```

```r
  ✖ add a "URL" field to DESCRIPTION. It helps users
    find information about your package online. If your
    package does not have a homepage, add an URL to GitHub, or
    the CRAN package package page.
  ✖ add a "BugReports" field to DESCRIPTION, and
    point it to a bug tracker. Many online code hosting
    services provide bug trackers for free,
    https://github.com, https://gitlab.com, etc.
```

Run `usethis::use_github_links()`.

```r
  ✖ avoid long code lines, it is bad for
    readability. Also, many people prefer editor windows that
    are about 80 characters wide. Try make your lines shorter
    than 80 characters

    R\authors_clean.R:24:1
    R\authors_clean.R:26:1
    R\authors_clean.R:29:1
    R\authors_clean.R:35:1
    R\authors_clean.R:36:1
    ... and 162 more lines
```

It's the complicated function, one more reason to try and simplify it?

```r

  ✖ omit trailing semicolons from code lines. They
    are not needed and most R coding standards forbid them

    R\authors_refine.R:20:198
```

```r

  ✖ avoid sapply(), it is not type safe. It might
    return a vector, or a list, depending on the input data.
    Consider using vapply() instead.

    R\authors_clean.R:97:17
    R\authors_clean.R:132:19
    R\authors_clean.R:152:19
    R\authors_clean.R:186:17
    R\authors_clean.R:223:22
    ... and 14 more lines

  ✖ avoid 1:length(...), 1:nrow(...), 1:ncol(...),
    1:NROW(...) and 1:NCOL(...) expressions. They are error
    prone and result 1:0 if the expression on the right hand
    side is zero. Use seq_len() or seq_along() instead.

    R\authors_clean.R:56:15
    R\authors_clean.R:71:79
    R\authors_clean.R:188:15
    R\authors_clean.R:209:21
    R\authors_clean.R:422:12
    ... and 9 more lines

  ✖ avoid 'T' and 'F', as they are just variables
    which are set to the logicals 'TRUE' and 'FALSE' by
    default, but are not reserved words and hence can be
    overwritten by the user.  Hence, one should always use
    'TRUE' and 'FALSE' for the logicals.

    R/authors_clean.R:NA:NA
    R/authors_clean.R:NA:NA
    R/authors_clean.R:NA:NA
    R/authors_clean.R:NA:NA
    R/authors_clean.R:NA:NA
    ... and 38 more lines
```

* Can you please re-trigger a Travis build so that the coverage badge indicate a coverage? We're aiming at a minimal coverage of 75% see https://github.com/ropensci/dev_guide/pull/94/files (brand-new official guidance)

* I'd recommend putting all badges on a single line.

* Could you also use Appveyor CI for Windows? `usethis::use_appveyor()`. This will add another badge.

* You can add the in-review badge

```
[![](https://badges.ropensci.org/256_status.svg)](https://github.com/ropensci/onboarding/issues/256)
```

It'll turn green when your package is approved.

* Could you please add examples in the documentation of the functions?

* Running `devtools::spell_check()` shows a few typos among the false negatives: querrying, nunmbers etc.

---

Reviewers:
Due date:

**Comment by aurielfournier** [comment]
Hi @maelle 

We are going to pause, and redo `authors_clean()` to be simpler/broken down into several functions. This will probably take ~ 1 week. 

Thanks!

**Comment by maelle** [comment]
Ok, thank you!

**Comment by maelle** [comment]
Thanks @aurielfournier and @birderboone!

A few more things from `goodpractice` to tackle before I look for reviewers

```r
It is good practice to

  
  ✖ add a "BugReports"
    field to DESCRIPTION, and point
    it to a bug tracker. Many
    online code hosting services
    provide bug trackers for free,
    https://github.com,
    https://gitlab.com, etc.
```
Simply run `usethis::use_github_links()`

```r
  ✖ use '<-' for
    assignment instead of '='. '<-'
    is the standard, and R users
    and developers are used it and
    it is easier to read your code
    for them if you use '<-'.

    tests\testthat\test_authors_match.R:4:5

```

The `styler` package might help.

```r

  ✖ avoid long code
    lines, it is bad for
    readability. Also, many people
    prefer editor windows that are
    about 80 characters wide. Try
    make your lines shorter than 80
    characters

    R\authors_address.R:12:1
    R\authors_address.R:14:1
    R\authors_address.R:17:1
    R\authors_address.R:20:1
    R\authors_address.R:41:1
    ... and 174 more lines
```


```r

  ✖ avoid sapply(), it is
    not type safe. It might return
    a vector, or a list, depending
    on the input data. Consider
    using vapply() instead.

    R\plot_net_address.R:32:26
    R\plot_net_address.R:33:26
    tests\testthat\test_references_read.R:10:17

  ✖ avoid 1:length(...),
    1:nrow(...), 1:ncol(...),
    1:NROW(...) and 1:NCOL(...)
    expressions. They are error
    prone and result 1:0 if the
    expression on the right hand
    side is zero. Use seq_len() or
    seq_along() instead.

    R\authors_georef.R:55:25
    R\authors_georef.R:71:15
    R\authors_georef.R:113:17
    R\plot_net_address.R:34:35
    R\plot_net_address.R:123:22
    ... and 1 more lines

  ✖ fix this R CMD check
    NOTE: Namespaces in Imports
    field not imported from:
    'Rdpack' 'maps' 'stringr' All
    declared Imports should be
    used.
  
  ✖ avoid 'T' and 'F', as
    they are just variables which
    are set to the logicals 'TRUE'
    and 'FALSE' by default, but are
    not reserved words and hence
    can be overwritten by the user.
    Hence, one should always use
    'TRUE' and 'FALSE' for the
    logicals.

    R/authors_address.R:NA:NA
    R/authors_address.R:NA:NA
    R/authors_address.R:NA:NA
    R/authors_georef.R:NA:NA
    R/authors_georef.R:NA:NA
    ... and 15 more lines
```

And from me: could you please add a coverage badge? `usethis::use_coverage()` should help you with that.

Thanks in advance and thanks for all your work until now! 😸

**Comment by aurielfournier** [comment]
Hi @maelle 

Thanks as always for your kind patience. It is greatly appreciated. 

I've addressed all of the above, and I finally downloaded `goodpractice` for myself to check things.

The one issue that I don't totally understand, but isn't throwing an issues in `goodpractice` is this one

` fix this R CMD check
    NOTE: Namespaces in Imports
    field not imported from:
    'Rdpack' 'maps' 'stringr' All
    declared Imports should be
    used.`

I thought that meant that I needed to remove `Rdpack`, `maps` and `stringr` from the DESCRIPTION file. So I did, but then the build failed, and it did not pass unless I put `Rdpack` back in.

But otherwise I think we're ok. :D

**Comment by maelle** [comment]
Thanks a lot @njahn82 @bmkramer for accepting to review this package! 😺  Your reviews are due on 2018-12-12.

As a reminder, our reviewer guide can be found [here](https://ropensci.github.io/dev_guide/onboarding-guide-for-reviewers.html) and the review template [here](https://ropensci.github.io/dev_guide/reviewtemplate.html).

**Comment by maelle** [comment]
:wave: @njahn82 @bmkramer! Friendly reminder that your reviews are due in two days, on 2018-12-12. 😺

**Comment by maelle** [comment]
Thanks a lot for your review @njahn82! 😺

**Comment by bmkramer** [comment]
## Package Review

*Please check off boxes as applicable, and elaborate in comments below.  Your review is not limited to these topics, as described in the reviewer guide*

- [x] As the reviewer I confirm that there are no [conflicts of interest](#coi) for me to review this work (If you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [x] **A statement of need** clearly stating problems the software is designed to solve and its target audience in README
- [x] **Installation instructions:** for the development version of package and any non-standard dependencies in README
- [x] **Vignette(s)** demonstrating major functionality that runs successfully locally
- [x] **Function Documentation:** for all exported functions in R help
- [x] **Examples** for all exported functions in R Help that run successfully locally
- [x] **Community guidelines** including contribution guidelines in the README or CONTRIBUTING, and DESCRIPTION with `URL`, `BugReports` and `Maintainer` (which may be autogenerated via `Authors@R`).

>##### For packages co-submitting to JOSS
>
>- [x] The package has an **obvious research application** according to [JOSS's definition](http://joss.theoj.org/about#submission_requirements)
>
>The package contains a `paper.md` matching [JOSS's requirements](http://joss.theoj.org/about#paper_structure) with:
>
>- [x] **A short summary** describing the high-level functionality of the software
>- [x] **Authors:**  A list of authors with their affiliations
>- [x] **A statement of need** clearly stating problems the software is designed to solve and its target audience.
>- [] **References:** with DOIs for all those that have one (e.g. papers, datasets, software).

*Comments: DOIs missing from a number of references; 'Larivière' spelled as 'Lariviare' on two occasions; figure not displaying correctly*

#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [ ] **Functionality:** Any functional claims of the software been confirmed.
*See comments listed below*
- [ ] **Performance:** Any performance claims of the software been confirmed.
*See comments listed below*
- [ ] **Automated tests:** Unit tests cover essential functions of the package
   and a reasonable range of inputs and conditions. All tests pass on the local machine.
*Comments:not done automated tests*
- [ ] **Packaging guidelines**: The package conforms to the rOpenSci packaging guidelines
*Comments: not feel qualified to assess*

#### Final approval (post-review)

- [ ] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**

Estimated hours spent reviewing: 6 hours

---

### Review Comments

General:

I really like the ability of this package to extract author and address information from WoS records. The main functionalities of the package worked (importing results, author name parsing and disambiguation and georeferencing) worked reasonably well for me.  The visualization functions I could not get to work. 

Information on functionality provided in vignette is detailed and complete; it would be good though to duplicate some of this (e.g. the examples) in Readme.md to get people started on how to use the package. 

I have focused my review on functionality of the package. Comments below are based on performing all tasks as described in the vignette, with a custom test dataset downloaded from Web of Science (first 500 articles from PeerJ in 2018)

_1 Introduction_
In describing the package, it is mentioned that the processed data-sets can be exported in tidy formats for more in-depth analyses with other packaged. It is not mentioned in what format export is useful for other packages. Perhaps this is self-explanatory (the csv-outputs provided), but it would be good to specify.

_2.0 Using Refnet_

- could not find test data ‘example_data.txt'
- refnet_fig1.jpg in Appendix 1 is very low res
- For WoS export: in text it is mentioned that both .txt and .ciw formats can be processed, but the worked example in Appendix 1 only shows how to export as .txt. Would be good to harmanize, and perhaps explain in text that .ciw is format for Endnote export
- In my tests, export does not need to be via Marked list, but also works via download menu in search results, either as Endnote export (.ciw) or as 'Other file formats', with 'full record' and 'plain text' selected. This works much faster than via Marked lists. 
 

_2.1 Importing Search Results_

- In references_read(), what is the default for dir=T/F? 
args(references_read) says it's TRUE

- Comments on vignette text:
-- typo in example a): .txr
-- c) is not a separate example
-- remark in Appendix 2 on fields only included when all.fiels=T is included in references_read() should be included in main text describing references_read()

- testing all.fields=T results in error: unused argument (all.fields=T)
args(references_read) reveals it should be include_all=FALSE

_2.2  Author address parsing and name disambiguation_

- Function authors_clean(): no csv file saved. 
Function contains argument write_out_data = FALSE. 
Tried TRUE => 2 files saved (authors_review, authors_prelim)
- Function also contains argument sim_score (value 0.88) - this is not explained in the documentation (it is mentioned for authors_refine where it has a NULL value)

- In documentation under 2.2.2, 
-- reference is made to Appendix 2, this should be Appendix 3.
-- it is stated: 'Users that prefer to manually review the results of the disambiguation can do so with the “authors” object and .csv files'
-> unclear which of the 2 csv files (prelim or review) should be taken (I assume review from the documentation of the next step. Also: 'authors' object is unclear)
- "Corrections made to the “review” file are merged into the “preview” file"-> should be "prelim"


- In Appendix 3: 
-- explanation on author name disambiguationis informative and useful. It does have some spelling and style issues, not critical, but could do with a careful edit
-- kudos for encouraging authors to sign up for (and use!) ORCID!
-- not covered:  sim_score
-- layout of table 2 (2 columns) is mangled
-- In table 2, similarity is listed as NA, but it has a value in my test data

_2.3 Georeferencing author institutions_

- Example for authors_georef is incomplete
example_georef <-authors_georef(------,------,-----)

- Explanations of arguments is incomplete: 
function (data, address_column = "address", filename_root = "", 
    write_out_missing = TRUE, retry_limit = 10) 
    
-> address_column has value
-> retry limit not discussed

- excuting authors_georef() resulted in lots of error messages, with a loop at the end and a message that a number of geocoding queries are remaining
-> if this is expected behaviour, would be good to address in documentation

- In documentation: not clear when which geocoding application is used when (sequentially?).
http://www.datasciencetoolkit.org/ and/or
https://developers.google.com/maps/documentation/.

- In documentation, it is stated 'an output/file of references that refnet was unable to georeference, which the user can review, manually correct, and import back into the file of georeferenced author locations
-> file seems to contain all lines (with and without lat/long resolved)
-> unclear how 'import back into file' should be performed

_2.4. Data Visualization: Productivity and Collaboration_

I could not get these to work, see error messages (and some analysis on them) below.

- Error in plot_addresses_country(PeerJ_2018_2_georef, filename_root = "./output/PeerJ_2018_2") : 
  unused argument (filename_root = "./output/PeerJ_2018_2")
  
- args(plot_addresses_country)
function (data, mapRegion = "world") -> so no argument for filename_root as in documentation

- Error in rworldmap::joinCountryData2Map(country_name_table, joinCode = "NAME",  : 
  your chosen nameJoinColumn :'country_name' seems not to exist in your data, columns = Freq 
  
- plot_addresses_points: also no argument for filename_root as in documentation

- Example from plot_net_coauthor() is incorrect:
plot_addresses_points <- plot_addresses_points(data, filename_root="./output/example")

> plot_net_coauthor_2 <- plot_net_coauthor(PeerJ_2018_2_georef)
Error in data[!is.na(data$country), ] : incorrect number of dimensions

> plot_net_country_2 <- plot_net_country(PeerJ_2018_2_georef)
Error in data[!is.na(data$country), ] : incorrect number of dimensions

- args(plot_net_country)
function (data, line_resolution = 10, mapRegion = "world")
-> so no argument for filename_root as in documentation

- Error in plot_net_addresses(PeerJ_2018_2_georef) : 
  could not find function "plot_net_addresses"

**Comment by bmkramer** [comment]
With apologies for the late review!

**Comment by maelle** [comment]
Thanks a lot for your review @bmkramer! :smile_cat: 

Reg "The package conforms to the rOpenSci packaging guidelines" the question is whether you see any discrepancy between https://ropensci.github.io/dev_guide/building.html and the package, if you have time you are qualified to assess, and you can ask me any question.

Was time or another problem the reason for not running tests? Happy to help if needed (well I can't help with time :smile: ).

Thanks a lot for your feedback in any case!

**Comment by maelle** [comment]
@aurielfournier @birderboone now both reviews are in! :tada:

**Comment by aurielfournier** [comment]
Just a note to all involved that @birderboone and I are working on the edits (huge thanks to the reviewers!), we're just a bit slowed down by other things at the moment, but we should have everything addressed by February 5th. Thanks for your patience!

**Comment by aurielfournier** [comment]
Thanks to the reviewers (@bmkramer , @njahn82) to providing several useful links to resource that made addressing their comments much easier! Your comments were very helpful and constructive and the package is much better off for them, we really appreciate your time! 

First, if you look at the repo, you will see that build is failing. This is because all of [this happened](https://github.com/dkahle/ggmap/issues/180), and basically if we were able to get travis to use the github version of ggmap, everything would be fine, but until those changes are on CRAN, the travis build will fail. Since our response was due today, and other then this issue we're ready for you all to look at it again I'm tossing the ball over to your side of the court. If you would like to wait till ggmap on CRAN is updated, and the build passes, that is fine by us. 

Below is our response broadly to the reviewer comments, if you would prefer a comment by comment response, let me know and I'm happy to do that. Thanks!

Auriel, Matt and Emilio 

~~

We are choosing at this time to not split up our functions anymore then they already are. We have split up the original functions into smaller pieces two times already.

While we appreciate the suggestion for using tidyverse functions, and we do use them in many other contexts, due to the changes in the tidyverse packages, that are not always backwards compatible, we have chosen to avoid them in many cases to avoid this package breaking because of that in the future.

We have changed the name of the package to refsplitr to avoid the conflict on CRAN

We have changed references_read to have dir default to FALSE, to help alleviate the issues the reviewer had

We removed all the csv writing outputs for functions where that was not needed as apart of the author refining process.

All typos and other small changes have been made, thank you to both reviewers for catching them

We have removed any need for the google API from the package, since between when we submitted and now it can no longer be used for free.

We have fixed the plotting functions to the best of our ability, some we were unable to replicate. If the reviewers find them again, can they share their input file so we can better diagnose the issue?


The reviewer is correct in that ciw formats can also be processed, and we have revised the text of the vignette to reflect that. We were initially reluctant to mention this because we wanted to avoid users download files in proprietary formats, but ciw files can be opened with a text editor).  We have also edited the Appendix showing how to download search results to include direct download of ciw files from the search results without going to marked list. 


Reviewer comment : In my tests, export does not need to be via Marked list, but also works via download menu in search results, either as Endnote export (.ciw) or as 'Other file formats', with 'full record' and 'plain text' selected. This works much faster than via Marked lists.

Response: This is indeed faster because it eliminates several steps. However, this will download all records resulting from a search, including any that were incorrectly returned (e.g., those by an author with an identical name). If users wish to filtering results prior to download to avoid including unwanted publications, then the best approach to save only the desired records to the Marked List and download from there as either a .ciw or .txt. Appendix 1 has been amended to include this option.

**Comment by maelle** [comment]
:wave: @aurielfournier @birderboone! Thanks for your answer.

>  if we were able to get travis to use the github version of ggmap
You can do that! See https://docs.travis-ci.com/user/languages/r/#remote-package :-)

@njahn82 @bmkramer thanks again for your reviews. Are you happy with the authors' response above?

**Comment by aurielfournier** [comment]
Thanks for the link @maelle 

I've added in the needed argument to the travis yml file, and the build still isn't working, though all the tests pass on my machine when I use the github version of ggmap, though it did take restarting everything to make that happen

So I'm not sure what is going on. :/

**Comment by aurielfournier** [comment]
Ok! I worked with some of the awesome ladies over in R-Ladies today, and we figured out the issue. 

Its actually an issue with `ggmap`. Jenny Bryan opened up an issue in [their repo about it. ](https://github.com/dkahle/ggmap/issues/264).

The solution that Jenny suggested was adding 
`options(ggmap = list(display_api_key = FALSE))` at the top of `authors_georef.R` and now the build is passing.  🎉

**Comment by maelle** [comment]
Awesome, well done you and Jenny!

**Comment by jennybc** [comment]
If you're going to set the option in this way, which seems reasonable for a semi-temporary workaround, you should technically be a bit more careful to put things back the way you found them. You only want your value of `FALSE` to hold for the duration of this function's execution.

At the place where you set the option, you could capture the existing value and immediately use `on.exit()` to schedule its restoration. Or you could use `withr::local_options()` accomplish both at once, with the downside that you'd need to Import withr.

**Comment by dkahle** [comment]
That problem should be fixed from __ggmap__'s side now (with 0c68d5c); let me know if that doesn't do it. Sorry for the problem!

**Comment by maelle** [comment]
@njahn82 @bmkramer thanks again for your reviews. Are you happy with the authors' response above?

**Comment by njahn82** [comment]
Sorry for my late reply. First of all thank you for your kind words and the changes you made. I am particularly impressed about your engagement with the R community to improve your work.

Before addressing the changes made, I wonder if I missed that runnable R code chunks were added to the README or vignette. As far as I see the vignette does not execute functions from the package, and there is no `README.Rmd` file. I am afraid it is formal requirement from rOpenSi that the vignette demonstrates that major functionality from the package runs successfully. As a user, I often look for such runnable examples before getting started with a package.

Can you point me to such document?

**Comment by aurielfournier** [comment]
Hi @njahn82 .

Perhaps I am misunderstanding the question, but ` refsplitr/vignettes/refsplitr-vignette.Rmd` contains chunks of code that can be run by the user which execute each function from the package. Which is a change we made in this last revision. 


for example: line 66 
```r
example_refs <- references_read(data = system.file("extdata", "example_data.txt", package = "refsplitr"),
                                    dir = FALSE)
```


Is this not what you meant? 

I also just updated the ReadMe file to have a the same example shown in the vignette.

**Comment by njahn82** [comment]
Sorry for the confusion. I thought of R code chunks indicated by curly brackets (```` ```{r} ````)  that are evaluated when a R Markdown document is rendered. The resulting output file shows the R output. In the vignette, it seems that package functions are highlighted (```` ```r ````). When rendered, no R output is presented, but screenshots from spreadsheet software.
Example: https://github.com/embruna/refsplitr/blob/16e7308fe75044e53848ab3bbecb80abb3cb7264/vignettes/refsplitr-vignette.Rmd#L99-L110

It would be be great to have some reproducible examples for the package's main functionalities.

**Comment by aurielfournier** [comment]
Agreed! We'll get right on it. Ggmap is giving us some issues again, but once we get those resolved we'll make those edits to the vignette and report back. 

Thanks for clarifying!

**Comment by aurielfournier** [comment]
Alright. We resolved the issue with ggmap. A vignette with rendered R output is now in the repo! Let me know if you have any other comments!

**Comment by maelle** [comment]
:wave: @bmkramer @njahn82, are you both happy with the authors' response?

**Comment by njahn82** [comment]
Unfortunately, I feel that some improvement is still needed.

It's great to have reproducible examples now in the vignettes. Sadly, I did not succeed building the vignette while installing the package. 

So I used the rendered [`refsplitr-vignette.html`](https://htmlpreview.github.io/?https://github.com/embruna/refsplitr/blob/master/vignettes/refsplitr-vignette.html) file instead: When describing `plot_net_country()` and `plot_net_address()`, it would be better to call the `$plot` element directly to avoid that the other list elements are printed out. It would be great to have an example how users can generate and customize their own plots using the other outputs provided by these functions.

`README.Rmd` needs to be added to the `.Rbuildignore` file to make the package more CRAN compatible. If it is intended to submit the package to CRAN, dependencies listed in `Remotes` must be available via CRAN. Otherwise, there will be a warning when running `R CMD check`.

As noted, ownership change of the Web of Science needs to be addressed; since 2016 the Web of Science has been provided by Clarivate Analytics, and not Thomson Reuters.

I also noted that functions could be more thoroughly documented. All functions lack `@examples` tags followed by example R code on how to use the function. See also <https://ropensci.github.io/dev_guide/building.html#examples>

Source code should adhere to a code style, especially spacing, to improve the readability of the source code <https://ropensci.github.io/dev_guide/building.html#code-style> . Practice checks
`goodpractice::gp()` and `lintr::lint()` help checking for good coding style.

Regarding the use of tidyverse, I am fine with not using it. However, as this package already makes heavy use of external packages including those from the tidyverse, I thought that it would make the programming of the package more coherent. 

Lastly, while playing around with the plotting functions, I wonder if you want to support transparent edges by default. Then, overlapping edges would become better visible. Here's an example:

Default:
<img width="767" alt="Bildschirmfoto 2019-04-01 um 14 29 59" src="https://user-images.githubusercontent.com/1702982/55329109-1f603a00-548e-11e9-8b60-1f7cd2a0ed4d.png">

With alpha transparency set to 0.1

![transparent_edges](https://user-images.githubusercontent.com/1702982/55329163-3a32ae80-548e-11e9-9215-0690d5fb79fa.png)

I also realized that `ggplot2::aes_string` is used, which is soft-deprecated. It is recommended to use tidy evaluation idioms instead. Would it be possible to update the ggplot2 functions accordingly?

**Comment by maelle** [comment]
Thanks @njahn82 for these useful review comments! @aurielfournier @birderboone could you please address those?

**Comment by aurielfournier** [comment]
Hi, Thanks @njahn82 for the comments. We'll get them addressed, it may be a bit delayed though as I'm on day 1 of two straight weeks of all day courses, but hopefully by the end of the month.

**Comment by aurielfournier** [comment]
Hi All, Matt and I are working on this, but its likely going to be mid May before we have everything pulled together. We apologize for the delay, and thank you for your patience, we're both doing this outside of our day jobs.

**Comment by maelle** [comment]
:wave: @aurielfournier & @birderboone! Thanks for the update, I understand.

**Comment by maelle** [comment]
wave @aurielfournier & @birderboone! Mid-May is now, any update? :wink:

**Comment by birderboone** [comment]
Hello,
So the package should be ready. 
- I have gone over it with lintr some more. I have not taken out upper case letters in variable names which lintr gets mentions frequently. I looked and I wasn't quite sure if this is mentioned in the ropensci/hadley style guide? I might not have looked hard enough. I didnt want to go down this path unless absolutely required since it would likely take a bit of time to make sure I found all instances of variable renames.  So let me know if its essential.
- I added @examples to every exported function, and included 2 data sets that are available to the users that are used in the examples
- I did add an alpha argument to the plot_net_country function because it did make it look much better. In the future I'll probably have to think of ways to make the wrapped ggplot functions available for extra tweaking as I'm sure people will have further style suggestions as the package gets used more.
- Additionally, the majority of the time required to finish these edits was due to changing the the internal structure of the package based on the other co-authors suggestions. So I apologize for the time delay. The outputs may have changed slightly due to these changes, however, the documentation should have changed to reflect those changes.

Thank you for your patience
- Matt

**Comment by maelle** [comment]
Thanks @birderboone! 

Regarding naming, the most important thing is to be consistent within the package, to make it easier for e.g. new contributors to pick things up.

@njahn82 does the response by the authors above address your concerns? Thanks in advance!

**Comment by njahn82** [comment]
Thank you! I lack time to look into the changing of the internal structure of the package. However, there are still some issues with the vignette. To speed things up, I sent an [pull request](https://github.com/embruna/refsplitr/pull/70), which addresses the following:

- YAML header uses standard vignette header
- refplitr is now imported from source, not from a local directory `("/home/matt/r_programs/refsplitr")`
- Fixed R Markdown syntax for headings
- Avoid loading pre-compiled images, using the output from the functions described in the vignette instead.
- fix encoding of a French author name in bibliography
- fix inline citation
- fix some typos

In the vignette, there is a non-runnable code chunk, which fails when executed:
https://github.com/embruna/refsplitr/blob/master/vignettes/refsplitr-vignette.Rmd#L192

Is it possible to support transparency in `plot_net_address()` as well?

I saw that `aes_string()` was changed to `aes_`. Unfortunately, this function will be soft-deprecated as well in the near future. It is recommended to use tidy evaluation idioms instead.

There are three warnings and two notes when the package is built using [travis](https://travis-ci.org/embruna/refsplitr/builds/537524725).

I do not want to be picky, but feel that fixing these issues will help to build trust in the functionalities of this useful package.

**Comment by maelle** [comment]
thanks a lot @njahn82! :smile_cat:

**Comment by aurielfournier** [comment]
Thanks @njahn82. We're getting a plan together soon to address these comments. I just moved across the country and am starting a new job, so it may take us a few weeks. Thanks again for everyone's patience.

**Comment by birderboone** [comment]
Hello!
I thank y'all for the constructive reviews throughout this process. I think the package  is 100% better than day 1 of submission .
Technical stuff:
-I did fix the tidy evaluation to enquo and !!, though I also just read the rlang update so I should also test out and then update to the double curly method. But i didn't want you to wait on me while I do that.
-I did add a linealpha argument to all network analysis. And we've been brain storming the best way to handle future ggplot tweaks a user might want to make. For tweaks that aren't as obvious as the alpha for the lines. Either coming up with a set of obvious plot tweak argument's, staying hands off, or something more elegant. 
-All Travis warnings  have been dealt with.

-The pull request for the vignette were very helpful. I merged most and only modified slightly some. So thank you so much for the editing suggestions. 

I think that was it.

-Matt

**Comment by njahn82** [comment]
Thank you for these changes, they look good to me. 

The vignette workflow still confuses me. I have the feeling that the vignette is not generated during the package installation process, but was compiled locally and was then moved from the vignette folder to `inst/docs`. Although this works, it may "shadow" package installation issues. Take for instances the dplyr package. In the vignette, dplyr is used, but not declared in the `DESCRIPTION` file. Also `knitr` is not declared.

Before we are entering another round, I would like to ask the editor @maelle if such pre-compiled vignettes comply with the rOpenSci guidelines.

**Comment by birderboone** [comment]
Honestly this may come down to my own confusion and lack of understanding. I am able to build the vignette locally with devtools, but when I push to github Travis gives me a warning that there is nothing in inst/doc. Which I just assumed meant they wanted a hard copy since devtools doesnt auto install vignettes anymore. I have no basis for that assumption just a guess I made, so I put a hardcopy in inst/doc

I can easily delete the inst/doc folder, but then I'll still get the Travis warning, perhaps there's something wrong in the yaml or something thats making the vignette misinstall on travis?

**Comment by njahn82** [comment]
Hi @birderboone I realized that the vignette builder was not properly declared in the DESCRIPTION. After deleting some files, I got it running. When installing via GitHub, please note that the vignette is not automatically built. I used the following command

```r
devtools::install_github("njahn82/refsplitr", build_vignettes = TRUE)
```

I'll send you a pull request with the changes made. Please have a careful look into it. It also seems that the package contains a few files and I am unsure whether they are needed or not.

**Comment by njahn82** [comment]
I tried installing `refsplitr` with [pak](https://github.com/r-lib/pak), which warns that the remote `lionel-/vdiffr` is not accessible. Would it be possible to use the version on CRAN instead? Maybe the R version referred to in the DESCRIPTION needs an upgrade, too?

**Comment by noamross** [comment]
👋 Hello all, just to let you know that I'll be the handling editor for the rest of this review process as Maëlle is on leave. 

@bmkramer, have the recent changes addressed your comments? (Aside from the vignette-building issue above)

**Comment by aurielfournier** [comment]
Glad to have you aboard @noamross ! Welcome 👋 !

Thanks @njahn82 for the pull request. @birderboone is super busy with his day job right now, so I'm working on the stuff you suggested. Hopefully I'll have it all addressed tonight, if not tomorrow. Thanks

**Comment by birderboone** [comment]
Everything should be working now. Had to wait for field work to be done to find the pesky problem with vignette loading. Thanks @njahn82 on your tweeks. vdiff-r is now on cran version, I forget why we had to switch to the remote at one point.

**Comment by njahn82** [comment]
Congrats @birderboone , glad that vignette loading works now. There's just a minor hint, To get rid of the last remaining NOTE message while checking the package build via travis, pleas add the README.Rmd file to `.Rbuildignore`. 

Otherwise, I am happy with the author's responses, as well as the current status of the package.

**Comment by aurielfournier** [comment]
Thanks @njahn82 . I got that changed made just now and the build passes.

**Comment by aurielfournier** [comment]
@noamross is there anything else you're waiting on from @birderboone or myself? Thanks!

**Comment by maelle** [comment]
I'm waiting for an answer from the other reviewer and if I don't get it I will do the checks myself this week or next. Thanks for your patience (and yes I'm back as volunteer editor, thanks @noamross for the help!)

**Comment by bmkramer** [comment]
Other reviewer here :-) Apologies for not responding to your email enquiry sooner @maelle  - I just returned from holiday. I will try to go through the updated package this weekend or next if that is still helpful. If that is taking things too long, I also understand of course.

**Comment by maelle** [comment]
Thanks a lot @bmkramer! Yes it would still be most helpful. :pray: And I hope you had a nice vacation!

**Comment by aurielfournier** [comment]
Hi @bmkramer 

Is this your review - https://github.com/embruna/refsplitr/pull/70#issuecomment-527953368

or is there more coming still? @birderboone and I just don't want to get ahead of ourselves if you are still working on your review. 

Thanks!

**Comment by maelle** [comment]
@bmkramer is the useful comment mentioned by @aurielfournier above your response to the changes in the package? If so could you please copy-paste it here? Thanks in advance and already many thanks for that comment!

**Comment by bmkramer** [comment]
Ah, yes, apologies for dropping off the face of the earth for a few days - and for somehow misplacing that comment. Copy-pasting below! 

As indicated, I looked at the package from a user perspective. Hope this complement the more structural/technical comments from @njahn82! 

------------------------------

Hi all,

Wow, following this process has shown me a lot about the large effort that goes into creating a full package, including making sure it builds right. Really impressed by all the work and commitment!

Going through the package's workflow from a user perspective, I noticed that there a still a few inconsistencies/omissions in the description of the functions in the vignette, mostly to do with the arguments listed and the code examples given. I've listed these below.

I was able to run all functionality with the test dataset (example_data.txt), including all visualizations. When I tested with a user-generated dataset from Web of Science (.txt file, 490 records), I encountered two instances that gave error messages:

In the step 'splitting author records' in authors_clean, the following error message was encountered at 48% progress and execution was halted:
Error in data.frame(names = unique(unlist(strsplit(C1_names, "; "))), :
row names contain missing values

When using only a subset of the file (1st 100 records), this error did not occur.

Running plot_net_country() gave the following error:
Error in Ops.data.frame(fromC, toC) :
‘+’ only defined for equally-sized data frames
Let me know if you want to have those files for testing (can't link them here b/c of WoS restrictions)

Below my comments on my observations on the vignette for the different functions in the package:

2.1 references_read()

differences in arguments listed:
-- arguments in vignette: data, dir, filename_root [NB filename_root gives error: unused argument]
-- arguments in vignette example: data, package, dir
-- arguments in help: data, dir, include_all
-- arguments in help example: data, package
example given in vignette does not correspond to recommended workflow for working with example data (creating directory 'data' and save the sample data ‘example_data.txt’ file in the ‘data’ folder)
Appendix 1

export to Endnote (ciw) is positioned as the preferred/primary export format, but example data are given as .txt)
export to .txt (Other file formats) is mentioned in step 4b only (not step 4)
export to other file formats requires some additional parameters to be set (record content, file
format)
2.2. authors_clean()

vignette states 2 arguments, while function seems to have only one (references)
argument for filename_root (or similar) is implied in the vignette text
make clear how to include output as csv
2.2.1-2.2.3 authors_refine()

In 2.2.1 vignette says function has 4 arguments, but only mentions review and prelim,
In 2.2.3, vignette does list all 4 arguments for authors_refine()
2.4.1 plot_addresses_country()

In vignette, argument mapRegion not mentioned
In vignette, no example code line provided
2.4.2 plot_addresses_points()

In vignette, argument mapCountry not mentioned
In vignette, not specified that argument data is addresses element of the authors_georef() object,
nor that that is NOT default in this function.
2.4.3 plot_net_coauthor()

vignette lacks example code line
2.4.5 plot_net_address()

only one of 3 additional arguments (lineAlpha) explained in vignette
Hope this is still helpful!

**Comment by maelle** [comment]
Thanks a lot @bmkramer!

@aurielfournier @birderboone any ETA on your answer? Thanks!

**Comment by aurielfournier** [comment]
Hi @maelle, 

Another member of our package building team, @embruna is working on them right now. He hopes to have all the edits completed by the end of the month. Thanks!

**Comment by embruna** [comment]
Hi all, I’m on it and hope to be done by wed or thursday.


Emilio
——————
Emilio M Bruna
University of Florida
www.BrunaLab.org<http://www.BrunaLab.org>

On Sep 19, 2019, at 2:39 AM, Maëlle Salmon <notifications@github.com<mailto:notifications@github.com>> wrote:


EXTERNAL EMAIL: Exercise caution with links and attachments.




Thanks a lot @bmkramer<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_bmkramer&d=DwMCaQ&c=sJ6xIWYx-zLMB3EPkvcnVg&r=hPRQO-QHqXhD8p9F43Mezw&m=uhLX5KQslxqukOU46MadFoBVcS9ynNCCmO-6UxtN0q4&s=jTTIwcCJOii-XiOcO5KokVJhYy9G6cFh_vZQ9AwMWek&e=>!

@aurielfournier<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_aurielfournier&d=DwMCaQ&c=sJ6xIWYx-zLMB3EPkvcnVg&r=hPRQO-QHqXhD8p9F43Mezw&m=uhLX5KQslxqukOU46MadFoBVcS9ynNCCmO-6UxtN0q4&s=PryY6Merx3tSfIIUU6FDw0IUQFWGzz3aFN_XH1q5w2A&e=> @birderboone<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_birderboone&d=DwMCaQ&c=sJ6xIWYx-zLMB3EPkvcnVg&r=hPRQO-QHqXhD8p9F43Mezw&m=uhLX5KQslxqukOU46MadFoBVcS9ynNCCmO-6UxtN0q4&s=rjM6Hwdg4Lu4QtmUR3iO76b1iVfMwYEBmarL62x3P5g&e=> any ETA on your answer? Thanks!

**Comment by embruna** [comment]
Hi @bmkramer, would you mind sending me a link to the WOS files that threw those errors (or the files themselves) so I can try and figure out what is going on? My email address is in my github profile.  Thanks!  

EB

**Comment by bmkramer** [comment]
Hi @embruna, you've got mail :-)

**Comment by embruna** [comment]
Hello all, a description of our changes in response to @bmkramer's excellent and thorough feedback.  

> I was able to run all functionality with the test dataset (example_data.txt), including all visualizations. When I tested with a user-generated dataset from Web of Science (.txt file, 490 records), I encountered two instances that gave error messages:
>
> a) In the step ‘splitting author records’ in authors_clean, the following error message was encountered at 48% progress and execution was halted:
> Error in data.frame(names = unique(unlist(strsplit(C1_names, “; “))), :
> row names contain missing values

Clarivate added a new field - Consortium Author (CA) - which is used when a paper is written by (or has as a coauthor) a research consortium or group.  In cases where there are no individuals named as authors - only the consortium author - there will be NAs in the AU and AF fields; this was the reason for the error. The authors_clean() function now checks for such cases first, and if any are discovered it stops the process and tells the user which records have a problem so they can be reviewed and corrected prior to running authors_clean(). The error message also provides guidance on how to do so.

> b) Running plot_net_country() gave the following error:
> Error in Ops.data.frame(fromC, toC) :
> ‘+’ only defined for equally-sized data frames

Rare country name variants can cause this errors. In this case it was because the country listing was “Serbia” instead of “Republic of Serbia”. We added this to the list of name variants the functions checks for prior to running plot_net_country().

> Below my comments on my observations on the vignette for the different functions in the package:
> 
> 2.1 references_read()
> 
> a) differences in arguments listed:
> a.1) arguments in vignette: data, dir, filename_root [NB filename_root gives error: unused argument]
> a.2) arguments in vignette example: data, package, dir
> a.3)  arguments in help: data, dir, include_all
> a.4) arguments in help example: data, package
> b) example given in vignette does not correspond to recommended workflow for working with example data (creating directory ‘data’ and save the sample data ‘example_data.txt’ file in the ‘data’ folder)

We have included all arguments and removed the text for the (deprecated) argument for the location in which to save .csv. (in an earlier version csv saving was the default, but we opted instead to make saving as a .csv optional. The revised vignette reflects this edit, and both the vignette and function show users how to save the output as a .csv if they wish). 

> Appendix 1
> a) export to Endnote (ciw) is positioned as the preferred/primary export format, but example data are given as .txt)
> b) export to .txt (Other file formats) is mentioned in step 4b only (not step 4)
> c) export to other file formats requires some additional parameters to be set (record content, file format)

We decided to streamline the guide by eliminating the steps for downloading directly from the search results and instead recommending that users add records to the Marked List. This was in part to simplify the figure, but also because Clarivate changed the location and menu structure for direct downloads from the search page.  The new figure explains records can be downloaded as either .txt or .ciw, without emphasizing one over the other.  

> 2.2. authors_clean()
> a) vignette states 2 arguments, while function seems to have only one (references)
> b) argument for filename_root (or similar) is implied in the vignette text
> c) make clear how to include output as csv

We have edited the vignette and examples. We originally had a second argument - the location to save a csv file - but then opted instead to make saving as a .csv optional. The revised vignette reflects this edit, and both the vignette and function show users how to save the output as a .csv if they wish. 

> 2.2.1-2.2.3 authors_refine()
> a) In 2.2.1: vignette says function has 4 arguments, but only mentions review and prelim,
> b) In 2.2.3: vignette does list all 4 arguments for authors_refine()

We have added descriptions of all arguments to the vignette and function help, and included all arguments in the examples for 2.2.1 and 2.2.3.

> 2.4.1 plot_addresses_country()
> a) In vignette, argument mapRegion not mentioned
> b) In vignette, no example code line provided
> 

We have included all arguments in the vignette, in the examples and function help, and now provide an example with all arguments. 

> 2.4.2 plot_addresses_points()
> a) In vignette, argument mapCountry not mentioned
> b) In vignette, not specified that argument data is addresses element of the authors_georef() object, nor that that is NOT default in this function.

We have included all arguments in the vignette and in the examples and function help. The vignette and help now specifies the assignment of the ‘data’ argument. For all functions we now indicate which arguments are optional and what the default argument values used in cases where they are not included.

> 2.4.3 plot_net_coauthor(): 
> a) vignette lacks example code line
> 

We have added an example code line.

> 2.4.5 plot_net_address()
> a) only one of 3 additional arguments (lineAlpha) explained in vignette

We have included all arguments in the vignette, in the examples and function help, and now provide an example with all arguments.

**Comment by maelle** [comment]
Thanks @embruna! @bmkramer, are you happy with the response above?

**Comment by bmkramer** [comment]
Yes, I am, thanks @embruna!

**Comment by maelle** [comment]
Thank you @bmkramer!

@embruna @aurielfournier @birderboone We're now closer to approval. :wink:

Some comments from me:

* I made a PR to remove the useless Maintainer field from the DESCRIPTION (the field is useless, not the maintainer :slightly_smiling_face: )
* The README could do with a bit more work to make it more attractive to potential users, which had been mentioned a bit earlier. In particular,
    * There's no plot, although the workflow has a plot command. And the workflow section could use a bit text.
    * I understand the need to inform people of the transfer from R-Forge but in my opinion it should come at the end of the README, because that's not what a potential user needs to see to be convinced (and they might stop reading sooner)
    * I really like @njahn82's summary at the beginning of his review "This is a specific package used for manipulating and analyzing authorship data from the Web of Science (WoS), a large toll-access literature and citation database indexing articles from around 12.000 academic journals. The packages imports local files that needs to be manually downloaded from the database. This is a quite common workflow when re-using WoS data, because API access is very costly and limited.", could (part of) it find a place in the README?
* Via goodpractice I get a test coverage of 65%, 

```r
> covr::package_coverage()
refsplitr Coverage: 65.91%
R/plot_net_address.R: 0.00%
R/plot_net_coauthor.R: 0.00%
R/plot_net_country.R: 0.00%
R/authors_clean.R: 73.81%
R/authors_georef.R: 75.00%
R/authors_refine.R: 81.82%
R/authors_parse.R: 86.76%
R/authors_address.R: 87.34%
R/plot_addresses_points.R: 89.74%
R/authors_match.R: 91.08%
R/split_names.R: 96.00%
R/references_read.R: 96.34%
R/plot_addresses_country.R: 100.00%
```

Why does the badge show a different percentage? Three functions have no tests, correct? The coveralls report seems to not be for the latest commit.

* Other goodpractice output

```r
✖ write short and simple functions. These
    functions have high cyclomatic complexity:authors_match
    (57).
```

Could it be simplified a bit?

```r
  ✖ avoid long code lines, it is bad for
    readability. Also, many people prefer editor windows
    that are about 80 characters wide. Try make your lines
    shorter than 80 characters

    R/authors_address.R:87:1
    R/authors_address.R:92:1
    R/authors_clean.R:3:1
    R/authors_clean.R:5:1
    R/authors_clean.R:7:1
    ... and 92 more lines
```

```r
  ✖ avoid 'T' and 'F', as they are just variables
    which are set to the logicals 'TRUE' and 'FALSE' by
    default, but are not reserved words and hence can be
    overwritten by the user.  Hence, one should always use
    'TRUE' and 'FALSE' for the logicals.

    R/authors_parse.R:NA:NA
    R/authors_parse.R:NA:NA
    R/authors_parse.R:NA:NA
    R/authors_parse.R:NA:NA
```

These are lines like https://github.com/embruna/refsplitr/blob/8e9e42904b3bda93c830664d162e01f49a8cb454/R/authors_parse.R#L58

* Output from `rcmdcheck::rcmdcheck(args = "--as-cran")`

```r
  
  Unknown, possibly mis-spelled, fields in DESCRIPTION:
    ‘Remotes’
```

I think you can use CRAN's ggmap now?
  
```r
  Found the following (possibly) invalid URLs:
    URL: https://CRAN.R-project.org/package=XXXXX
      From: inst/doc/refsplitr-vignette.html
      Status: 404
      Message: Not Found
    URL: https://cran.r-project.org/web/packages/bibliometrix/index.html
      From: inst/doc/refsplitr-vignette.html
      Status: 200
      Message: OK
      CRAN URL not in canonical form
    URL: https://cran.r-project.org/web/packages/revtools/index.html
      From: inst/doc/refsplitr-vignette.html
      Status: 200
      Message: OK
      CRAN URL not in canonical form
    The canonical URL of the CRAN page for a package is 
      https://CRAN.R-project.org/package=pkgname
```

I didn't make this rule but we need to obey it.

```r

❯ checking examples ... NOTE
  Examples with CPU or elapsed time > 5s
                     user system elapsed
  plot_net_address 10.001  0.032  10.611
  plot_net_country  6.235  0.028   6.276
```

When CRAN submission approaches keep this in mind.

**Comment by maelle** [comment]
@embruna @aurielfournier @birderboone any update? Thank you 🙂

**Comment by maelle** [comment]
@embruna @aurielfournier @birderboone could you please respond soon? Thanks. :smile_cat:

**Comment by aurielfournier** [comment]
Sorry! @embruna is taking the lead on this next round of edits and he is working on it, but it will be a bit still. @embruna is a professor at U Florida and its getting to be his busy time of the semester. Thanks for your patience.

**Comment by maelle** [comment]
Thanks for the update!

**Comment by embruna** [comment]
Yes, I'm on it! Between my new class this semester, travel, and...well, the same as the rest of us. I'm on it.

**Comment by embruna** [comment]
Hello, a quick update. We are almost done - here is a summary of the changes made in response to feedback from @maelle. 

```I made a PR to remove the useless Maintainer field from the DESCRIPTION (the field is useless, not the maintainer 🙂 )``` Accepted.

```The README could do with a bit more work to make it more attractive to potential users, which had been mentioned a bit earlier. In particular, There's no plot, although the workflow has a plot command.```. We added one, and I defintiely think it makes it look more attractive. :)

```And the workflow section could use a bit text.``` I added a little, though I was concerned I would just be duplicating what was in the vignette.  

```I understand the need to inform people of the transfer from R-Forge but in my opinion it should come at the end of the README, because that's not what a potential user needs to see to be convinced (and they might stop reading sooner)```. Good suggestion, now at the bottom. ACCEPTED

```I really like @njahn82's summary at the beginning of his review "This is a specific package used for ...API access is very costly and limited.", could (part of) it find a place in the README?``` We included this, and it does nicely emphasize one reason why it's needed (cost).

- All the lines of code identified as being > 80 chrs are now < 80 chrs.
- All the instances of ```= F``` corrected to ```= FALSE``` 
- URLs in the vignettes now in CRAN canonical form.

The main thing I think we have left to deal with is the "Badge shows different % coverage", though we think that's a minor thing. We'll have an update on that ASAP. Thanks so much for the patience!

**Comment by maelle** [comment]
Thanks you! 😀

I think you forgot to upload the README image to the repo. 😉

The coverage difference isn't that minor because 65% is below the required 75% https://devguide.ropensci.org/building.html#testing

**Comment by maelle** [comment]
Reg overlap between the README and vignette you can re-use content without copy pasting if you want see https://www.garrickadenbuie.com/blog/dry-vignette-and-readme/

Moreover with roxygen2 latest version one can even include Rmd in man pages so you could re-use content even more (e.g. in the package level man page) https://roxygen2.r-lib.org/articles/rd.html#including-external--rmd-md-files

Repetition between docs is fine since one never knows what an user finds first.

**Comment by embruna** [comment]
> Thank you! 😀
> 
> I think you forgot to upload the README image to the repo. 😉
> 
> The coverage difference isn't that minor because 65% is below the required 75% https://devguide.ropensci.org/building.html#testing

ARGH! I did indeed. It should be there now.  

Still looking into the coverage issue.

**Comment by birderboone** [comment]
Ok everything should be ready to go:
- Coverall issue has been fixed. When we did the name change from refnet -> refsplittr coveralls was not properly configured to pull from the new names travis account.

- Coverage statistics are running now at...i think 91.6%. For some reason the tests for the plots had skip_on_cran() which the covr package skips for test coverage as well. Thats fixed, and I added a couple extra tests.

- I did some testing and ggmaps::geocode on CRAN is still not fixed. I have that the latest cran update was in February, and that version has the issue still. In ggmap::geocode this line needs to be commented out
  > if (source == "dsk") stop("datasciencetoolkit.org terminated its map service, sorry!")

Which it **is** in the github version after our issue submission, but has not been pushed to cran. Theres a lot of things that bother me about this whole situation, and hopefully datasciencetoolkit does not in the future shut down, otherwise we'll have to go a different route, or allow people to enter there google api keys (NOT IDEAL).

- We have not at this time decided to split up authors_match(). This is the core of what this package is, and its a complicated piece of code to start tweaking. We are in agreement with you that it does need to be paired down and split into smaller functions, but at this moment doing so is a large time investment. So we're currently tabling it for 2.0.  

Thanks for everything
- Matt

**Comment by maelle** [comment]
Thanks! We're getting very close to approval! I have a few remaining comments 

* from myself

    * I'd recommend changing the GitHub repo description to something more directly informative (as opposed to "update to")

    * In the Travis config file, it's not a good idea to ignore warnings now that the package is in such good shape.

https://github.com/embruna/refsplitr/blob/e1492dac22ac7b553b84f715c31e01086137be83/.travis.yml#L6

* from `goodpractice::gp()`


```r

  ✖ avoid long code lines, it is bad for
    readability. Also, many people prefer editor
    windows that are about 80 characters wide. Try
    make your lines shorter than 80 characters

    R/authors_address.R:89:1
    R/authors_refine.R:3:1
    R/authors_refine.R:4:1
    R/authors_refine.R:6:1
    R/authors_refine.R:9:1
    ... and 71 more lines
```

I see that e.g. the first one is long but ok as is, in this case you could add `# nolint` at the end. See https://github.com/jimhester/lintr#project-configuration 

```r
  ✖ fix this R CMD check NOTE: Namespaces
    in Imports field not imported from: ‘Rdpack’
    ‘gdtools’ ‘mapproj’ ‘vdiffr’ All declared
    Imports should be used.
```

Can you fix DESCRIPTION?

**Comment by embruna** [comment]
Hi @maelle, I think we got all of these taken care of.

- A few of the lines of code are at 81 characters, but I couldn't reduce them further because of the required formatting.  I hope that's ok. 

- The description on the repo has been changed to eliminate the "update to refnet text"

-Coverage 91% !  :)

**Comment by maelle** [comment]
Hi @embruna, thanks! Np reg the remaining long lines.

Reg DESCRIPTION could you have a look at dependencies ‘Rdpack’ ‘gdtools’ ‘mapproj’ ‘vdiffr’ as noted in my previous comment? Dependencies only used in tests should be under Suggests. by the way the line for Imports is very long, I'd **suggest** running `desc::desc_normalize()` to have one line per dependency (like what you have for Suggests), which would be easier to browse (that function also orders dependencies alphabetically).

Reg Travis what I meant is that I recommend undoing this commit https://github.com/embruna/refsplitr/commit/92e9fe5eff948f54ae9892526ddfdf83e322c461 since your package now is in very good shape, it'd be good for Travis to flag new WARNINGs as they appear.

After that we'll be done. Thanks for your patience and thanks again for your work!

**Comment by embruna** [comment]
OK, drum roll...

1. I moved the packages to the Suggests, and ran ```desc::desc_normalize()``` on the DESCRIPTION. The packages are also listed in alphabetical order.

2. travis.yml now has ```warnings_are_errors: true```

We're the ones who are grateful, both for your patience and the attention to detail. The review of this package has been incredibly helpful and detailed....I wish my papers were reviewed this thoroughly!

**Comment by maelle** [comment]
Thank you! :grin: And thanks again to both reviewers!

I'm sorry but I now realize these packages are maybe not even used at all in your package, or am I missing something? I've searched for their name in the package code (e.g. https://github.com/embruna/refsplitr/search?q=mapproj&unscoped_q=mapproj&type=Code)
* mapproj
* Rdpack I suppose you used to use it for package docs, but now you no longer use it?
* vdiffr I see https://github.com/embruna/refsplitr/blob/3ba804f62014aee01c370e5e14209f8c2817cf73/tests/figs/deps.txt but right now there doesn't seem to be any test using vdiffr, the tests in https://github.com/embruna/refsplitr/blob/master/tests/testthat/test_plots.R don't use it.

If the packages are really not used, could you please remove them from Suggests?

**Comment by maelle** [comment]
You don't need covr either since it's run on Travis. I've now found a way to find unneeded dependencies using the [attachment package](https://thinkr-open.github.io/attachment/) from inside the refsplitr folder.

``` r
desc_deps <- attachment::att_from_description()
scripts_deps <- attachment::att_from_rscripts(".")
namespace_deps <- attachment::att_from_namespace()
vignette_deps <- attachment::att_from_rmds()
desc_deps[!desc_deps %in% c(scripts_deps, namespace_deps, vignette_deps)]

```

**Comment by maelle** [comment]
Note that you could still have false positives since you could use Suggests packages for other things.

**Comment by maelle** [comment]
Happy New Year!

@embruna Once you've responded to my comment about Suggests unneeded dependencies the package will be approved. 🙂

**Comment by embruna** [comment]
Hi all, happy New Year.

Deleted the 4 packages from the Suggests....thanks @maelle!

**Comment by embruna** [comment]
Well, it was too good to be true...just got this email: 
![image](https://user-images.githubusercontent.com/4655487/71922599-3e94a200-3159-11ea-87c3-78b1748fdbe1.png)

I'm going to try adding the four packages to suggests one at a time and seeing which of them actually do need to be there.  I'll update shortly.

**Comment by maelle** [comment]
Ah indeed sorry mapproj seems needed https://travis-ci.org/embruna/refsplitr/jobs/633881379#L6325

**Comment by embruna** [comment]
``` Quitting from lines 342-344 (refsplitr-vignette.Rmd) 
6324
Error: processing vignette 'refsplitr-vignette.Rmd' failed with diagnostics:
6325
there is no package called 'mapproj' ```

**Comment by embruna** [comment]
Yes, I just reinserted to the Suggests. Fingers crossed!

**Comment by maelle** [comment]
Cool! I'll come back to this thread tomorrow, it's getting late here.

**Comment by maelle** [comment]
Approved! Thanks @embruna @aurielfournier @birderboone for submitting and @njahn82 @bmkramer for your reviews! :grin:

To-dos:
- [ ] Transfer the repo to rOpenSci's "ropensci" GitHub organization under "Settings" in your repo.  I have invited you to a team that should allow you to do so.  You'll be made admin once you do.
- [ ] Add the rOpenSci footer to the bottom of your README
"```
[![ropensci_footer](https://ropensci.org/public_images/ropensci_footer.png)](https://ropensci.org)```"
- [ ] Fix all links to the GitHub repo to point to the repo under the ropensci organization.
- [ ] If you already had a `pkgdown` website, fix its URL to point to `https://docs.ropensci.org/package_name` and deactivate the automatic deployment you might have set up, since it will not be built centrally like for all rOpenSci packages, see https://devguide.ropensci.org/ci.html#even-more-ci-ropensci-docs. In addition, in your DESCRIPTION file, include the docs link in the `URL` field alongside the link to the GitHub repository, e.g.: `URL: https://docs.ropensci.org/foobar (website) https://github.com/ropensci/foobar`
- [ ] Add a mention of the review in `DESCRIPTION` via [`rodev::add_ro_desc()`](https://docs.ropensci.org/rodev/reference/add_ro_desc.html).
- [ ] Fix any links in badges for CI and coverage to point to the ropensci URL. We no longer transfer Appveyor projects to ropensci Appveyor account so after transfer of your repo to rOpenSci's "ropensci" GitHub organization the badge should be `[![AppVeyor Build Status](https://ci.appveyor.com/api/projects/status/github/ropensci/pkgname?branch=master&svg=true)](https://ci.appveyor.com/project/individualaccount/pkgname)`.
- [ ] We're starting to roll out software metadata files to all ropensci packages via the Codemeta initiative, see https://github.com/ropensci/codemetar/#codemetar for how to include it in your package, after installing the package - should be easy as running codemetar::write_codemeta() in the root of your package.
- [ ] Activate Zenodo watching the repo
- [ ] Tag and create a release so as to create a Zenodo version and DOI
- [ ] Submit to JOSS at <https://joss.theoj.org/papers/new>, using the rOpenSci GitHub repo URL. When a JOSS "PRE REVIEW" issue is generated for your paper, add the comment: `This package has been reviewed by rOpenSci: https://LINK.TO/THE/REVIEW/ISSUE, @maelle`

Should you want to acknowledge your reviewers in your package DESCRIPTION, you can do so by making them `"rev"`-type contributors in the `Authors@R` field (with their consent).  More info on this [here](https://devguide.ropensci.org/building.html#authorship).

Welcome aboard! We'd love to host a blog post about your package - either a [short introduction to it with one example](https://ropensci.org/tech-notes/) or [a longer post with some narrative about its development or something you learned, and an example of its use](https://ropensci.org/blog/). If you are interested, review [the instructions](https://github.com/ropensci/roweb2#contributing-a-blog-post), and tag   `@stefaniebutland` in your reply. She will get in touch about timing and can answer any questions.

We've put together an online book with our best practice and tips, [this chapter](https://devguide.ropensci.org/collaboration.html) starts the 3d section that's about guidance for after onboarding. Please tell us what could be improved, the corresponding repo is [here](https://github.com/ropensci/dev_guide).

**Comment by embruna** [comment]
Hi All, a quick update:

The following have been COMPLETED
- Transfer the repo to rOpenSci's "ropensci" GitHub

-  Add the rOpenSci footer to the bottom of your README

-  Fix all links to the GitHub repo to point to the repo under the ropensci organization.

-  Fix any links in badges for CI and coverage to point to the ropensci URL. 

I'm finishing the following as soon as I am done with class:

- Add a mention of the review in DESCRIPTION via rodev::add_ro_desc().
- Including Metadata 
- I'm awaiting approval from one last reviewer

At that point I will create the Zeonodo 1.0 version and mint the DOI, and submit to JOSS. Should be done shortly.

**Comment by embruna** [comment]
Hi @noamross and @maelle, a couple quick questions:

1) I've written some code that helps users more quickly narrow down the names to review after doing the disambiguation, but we decided not to include it in the package.  I have it on my own github [(link here)](https://github.com/embruna/refsplitr_simplify_authors_review) and point to it in the vignette. Is that ok, or should this also be transferred over to the ROpenSci repo? 

2) It's currently versioned as v0.6. Should it be posted as v1.0?

3) I'm just waiting to hear back from a reviewer. I'll wait until Monday noon, at which time I'll do the  Zenodo stuff and submit to JOSS.

**Comment by maelle** [comment]
1. Maybe it could become a tech note (https://ropensci.org/technotes), or a vignette in refsplitr?

2. That's up to you. :slightly_smiling_face: 

3. Ok!

**Comment by embruna** [comment]
OK, @noamross and @maelle: refsplitr should be ready for Zendo v1.0.0

The only unexpected thing that came up was that our title (or rather, the text after the colon in `refsplitr: ---` needed to be fixed. It was inconsistent within the repo (different in readme and description) and far longer than CRAN-style.  So it is now as follows: 

`refsplitr: author name disambiguation, author georeferencing, and mapping of coauthorship networks with Web of Science data.`

I was a actually a bit concerned because Clarivate's webpage is clear: 

> Do not incorporate any Clarivate Analytics product names into your company's product names. [link](https://clarivate.ru/legal/trademark-notice/)

However, a quick search on CRAN reveals several other packages wth "Web of Science" in the title, so I just left it in. If you prefer it be removed let me know. Otherwise, I will freeze v1.0.0 with Zenodo at the end of the day.

**Comment by maelle** [comment]
I think the package isn't a company product? But I'm not a lawyer.

In any case I think you'll need to write it `'Web of Science'` in DESCRIPTION.

**Comment by embruna** [comment]
Submitted to JOSS, and Zenodo archived as v0.9.0 with doi 10.5281/zenodo.3608285. Once the JOSS review is complete and I can include the details in the paper.rm file. At that point I'll freeze v1.0.   A minor thing, and maybe unnecessary. But I wanted 1.0 to be a complete package with no details to be filled in.

**Comment by maelle** [comment]
Ok! I don't see the submission at https://joss.theoj.org/papers yet.

**Comment by embruna** [comment]
Yes, still listed as "pending"...I'll check on it periodically throughout the day.

**Comment by maelle** [comment]
I've just checked again. Please ping me if you see it appear. :wink:

**Comment by maelle** [comment]
Paper accepted at JOSS 🎉 so closing this issue.
